python_modules/dagster/dagster/_core/execution/plan/inputs.py

```diff
@@ -976,19 +979,79 @@ def __new__(
             input_name=check.opt_str_param(input_name, "input_handle", default=""),
         )
 
-    @property
-    def resolved_by_step_key(self) -> str:
-        return self.unresolved_step_output_handle.resolved_by_step_key
+    def get_resolving_handle(self):
+        return self.unresolved_step_output_handle.resolution_source.get_resolving_handle()
 
-    @property
-    def resolved_by_output_name(self) -> str:
-        return self.unresolved_step_output_handle.resolved_by_output_name
+    # @property
+    # def resolved_by_step_key(self) -> str:
+    #     return self.unresolved_step_output_handle.resolved_by_step_key
+
+    # @property
+    # def resolved_by_output_name(self) -> str:
+    #     return self.unresolved_step_output_handle.resolved_by_output_name
 
-    def resolve(self, mapping_key: str) -> FromStepOutput:
+    def resolve(self, mapping_key: str) -> Union[FromStepOutput, FromPendingDynamicStepOutput]:
```

Comment:
reading through this again before i try to merge it with my changes - should this type annotation be `Union[FromStepOutput, FromUnresolvedStepOutput]` or should the else condition return a `FromPendingDynamicStepOutput`?

Reply:
I think this one should be `Union[FromStepOutput, FromUnresolvedStepOutput]` and below [2] probably needs to be `Union[FromPendingDynamicStepOutput, FromUnresolvedDynamicStepOutput]`